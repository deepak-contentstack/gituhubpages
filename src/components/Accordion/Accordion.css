@import '../../../tokens/variables.scss';
@import '../../../tokens/tokens.css';
@import '../../static/styles/css/common.css';

.Accordion {
    font-family: $font-family-primary;

    &--backgroundColor {
        .Accordion {
            &__heading {
                padding-left: $space-10;
                &:hover {
                    background-color: $color-brand-white-light;
                }
            }
            &__open {
                margin-left: $space-10;
            }
            &__data {
                .Accordion {
                    &__open {
                        margin-left: 0;
                    }
                }
            }
        }
    }

    &__actions {
        align-items: center;
        background: $color-brand-white-base;
        border-radius: $radii-6;
        box-shadow: 0 $space-8 $space-20 rgba(34, 34, 34, 0.1);
        display: flex;
        height: $space-28;
        margin-right: $space-10;
        visibility: hidden;
        .actions_list_item {
            align-items: center;
            cursor: pointer;
            display: flex;
            height: 100%;
            justify-content: center;
            width: $space-28;
            &:hover {
                background-color: $color-brand-secondary-lighter;
                border-radius: $radii-6;
                svg {    
                    path {
                        fill: $color-base-gray-20;
                    }
                }       
            }    
        }
    }

    &__heading {
        align-items: center;
        border-radius: $radii-4;
        cursor: pointer;
        display: flex;
        height: $space-40;
        
        & > .flex {
            max-width: 70%;
        }
        &.Accordion__heading-disabled{
            cursor: not-allowed;
        }

    &__title {
      color: $color-font-secondary;
      font-size: $size-font-small;
      font-style: normal;
      font-weight: $font-weight-semi-bold;
      letter-spacing: -0.0025em;
      line-height: $space-14;
      &__ellipsis {
        @include text-wrapper;
      }
    }

        &:hover {
            .Accordion__heading__line,.Accordion__actions {
                visibility: visible;
            }
            .Accordion__heading__toggle {
                background: $color-base-gray-10;
                svg {
                    path {
                        fill: $color-base-white-5;
                    }
                }
            }
            .Accordion__heading__toggle-disabled {
                cursor: not-allowed;
            }
            .Accordion__heading__title {
                color: $color-font-black;
            }
        }

        &__field-count {
            align-items: center;
            background: $color-font-quaternary;
            border-radius: $radii-4;
            color: $color-font-base;
            display: flex;
            font-size: $size-font-small;
            font-weight: $font-weight-bold;
            height: $space-20;
            justify-content: center;
            margin-left: $space-10;
            min-width: $space-22;
            padding: $space-2 0.188rem;
        }

        &__toggle, &__toggle-disabled {
            align-items: center;
            border-radius: $radii-50;
            display: flex;
            height: $space-28;
            justify-content: center;
            margin-left: $space-10;
            min-width: $space-28;
        }

        &__errorMsg {
            color: $color-base-orange-5;
            font-size: $size-font-small;
            font-weight: $font-weight-bold;
            margin-left: $space-10;
        }

        &__line {
            flex-grow: 1;
            margin: 0 $space-10;
            visibility: hidden;
        }
    }

    &__collapsed {
        border-radius: $radii-8;
        height: 0;
        overflow: hidden;
    }

    &__open {
        height: auto;
    }

    &__data {
        border-left: 1px solid rgba(113, 128, 150, 0.3);
        font-size: $size-font-small;
        padding-top: $space-5;

        .Accordion {
            margin-left: 0;
            &__data {
                border-left: 1px solid rgba(113, 128, 150, 0.3);
                margin-left: $space-10;
            }

            &__heading {
                padding-left: $space-10;
            }
        }
    }
}
