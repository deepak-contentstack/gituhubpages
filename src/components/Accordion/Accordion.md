# Accordion

## Best practices
- Avoid the Accordion data to be pre-rendered for the initial render of the accordion. It will be available on the DOM once the data is loaded in the accordion and the preRender will be automatically set to `true`.