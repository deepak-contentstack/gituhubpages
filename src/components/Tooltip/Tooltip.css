@import '../../../tokens/variables.scss';
@import '../../../tokens/tokens.css';
@import '../../static/styles/css/typography.css';
@import '../../static/styles/css/variable.css';
@import '../../static/styles/css/common.css';

@mixin font-Inter {
    font-family: $font-family-primary;
}

.tippy-box[data-animation='fade'][data-state='hidden'] {
    opacity: $opacity-0;
}

.tippy-box[data-inertia][data-state='visible'] {
    transition-timing-function: cubic-bezier(0.54, 1.5, 0.38, 1.11);
}

.tippy-wrapper{
    display: inline-block;
}

.tippy-box-dark {
    background-color: $color-brand-secondary-dark;
    border-radius: $radii-4;
    color: $color-font-white;
    cursor: pointer;
    font-size: $size-font-small;
    font-weight: $font-weight-medium;
    letter-spacing: 0.04em;
    min-height:$space-22;
    transition-property: transform,visibility,opacity;
    @include font-Inter;
    
    >.tippy-content {
        line-height: $space-15;
        padding: $space-4 $space-8;
    }
}

.tippy-box-dark-tooltip {
    background-color: $color-brand-secondary-dark;
    border-radius: $radii-4;
    color: $color-font-white;
    cursor: pointer;
    font-size: $size-font-small;
    font-weight: $font-weight-medium;
    letter-spacing: 0.04em;
    min-height:$space-22;
    transition-property: transform,visibility,opacity;
    @include font-Inter;
    position: absolute;
    top: -30px;
    left: 0px;
    width:max-content;
    

    >.tippy-content {
        line-height: $space-15;
        padding: $space-4 $space-8;
    }
}

.tippy-box-light {
    background-color: $color-brand-white-base;
    border-radius: $radii-10;
    box-shadow: 0 4px 30px rgba(0, 0, 0, 0.25);
    color: $color-font-base;
    cursor: pointer;
    font-size: $size-font-small;
    font-weight: $font-weight-medium;
    letter-spacing: -6e-05em;
    text-align: left;
    @include font-Inter;

    &--primary {
        max-height: 22.81rem;
        max-width: 25rem!important;
        overflow: auto;
        padding: calc(var(--space-16) + 1px) $space-20;
        width: 25rem;

        > .tippy-content {
            .tippy-card {
                border-bottom: 1px solid $color-border-lighter;
                margin-bottom: $space-20;
                padding-bottom: $space-8;
                position: relative;

                .Env-label {
                    color: $color-font-base;
                    font-size: $size-font-small;
                    font-weight: $font-weight-medium;
                    margin-left: $space-5;
                }

                .tippy-card-content {
                    padding: $space-18 0 $space-18 $space-15;

                    .tippy-label {
                        color: $color-font-base;
                        font-size: $size-font-small;
                        font-weight: $font-weight-bold;
                    }
                    .tippy-value {
                        color:$color-font-tertiary;
                        font-size: $size-font-small;
                        font-weight: $font-weight-medium;
                        letter-spacing: -6e-5em;
                        margin-top: $space-5;
                    }
                    .tippy-lang {
                        color: $color-font-base;
                        font-size: $size-font-small;
                        font-weight: $font-weight-regular;
                    }
                }
                &:last-child {
                    border-bottom: 0;
                    margin-bottom: 0;
                    padding-bottom: 0;
                }
            }
        }
    }

    &--secondary {
        max-width: 22.5rem !important;
        padding: $space-15 $space-20;
        width: auto;

        >.tippy-content {
            line-height: $space-15;
            .tippy-content-title {
                color:$color-font-base;
                font-weight: $font-weight-bold;
                margin-bottom: $space-6;
            }
        }
    }


    &--dynamic {
        padding: $space-15 $space-20;
        > .tippy-content {
            line-height: $space-15;
            .tippy-content-title {
                color: $color-font-base;
                font-weight: $font-weight-bold;
                margin-bottom: $space-6;
            }
        }
    }
}

.TooltipWrapper {
    padding:  3.125rem 6.25rem;
}

.KeyShortcut {
    background: $color-brand-primary-base;
    border-radius: $space-5;
    color: $color-font-white;
    font-size: $size-font-tiny;
    font-weight: $font-weight-bold;
    height: $space-20;
    margin-left: $space-10;
    padding: $space-5 0;
    text-align: center;
    width: $space-20;
}

.tippy-box-light-menu {
    border-radius: $radii-6;
    box-shadow: 0 4px 10px rgba(0, 0, 0, 0.1);
    color: $color-font-base;
    font-size: $size-font-small;
    font-weight: $font-weight-medium;
    padding: $space-6 $space-10;
    position: relative;
    right: 1px;
    transition: 0ms !important;
    white-space: nowrap;
    .tippy-content {
        transition: 0ms !important;
    }
}

.tippy-arrow-light {
    border: $radii-8 solid transparent;
    height: 0;
    position: absolute;
    width: 0;
    
    &-left{
        align-self: center;
        border-right: $radii-8 solid $color-brand-white-base !important;
        left: -$space-16
    }

    &-left-start{
        align-self: flex-start;
        border-right: $radii-8 solid $color-brand-white-base !important;
        left: -$space-16
    }

    &-left-end{
        align-self: flex-end;
        border-right: $radii-8 solid $color-brand-white-base !important;
        left: -$space-16
    }
    
    &-right{
        align-self: center;
        border-left: $radii-8 solid $color-brand-white-base !important;
        right: -$space-16
    }

    &-right-start{
        align-self: flex-start;
        border-left: $radii-8 solid $color-brand-white-base !important;
        right: -$space-16
    }

    &-right-end{
        align-self: flex-end;
        border-left: $radii-8 solid $color-brand-white-base !important;
        right: -$space-16
    }

    &-top{
        border-bottom: $radii-8 solid $color-brand-white-base !important;
        left: calc(50% - var(--space-8));
        top: -$space-16;
        
    }

    &-top-start{
        border-bottom: $radii-8 solid $color-brand-white-base !important;
        left: calc(20% - var(--space-8));
        top: -$space-16;
    }

    &-top-end{
        border-bottom: $radii-8 solid $color-brand-white-base !important;
        right: calc(20% - var(--space-8));
        top: -$space-16;
    }
    
    &-bottom{
        border-top: $radii-8 solid $color-brand-white-base !important;
        bottom: -$space-16;
        left: calc(50% - var(--space-8));
    }

    &-bottom-start{
        border-top: $radii-8 solid $color-brand-white-base !important;
        bottom: -$space-16;
        left: calc(20% - var(--space-8));
    }

    &-bottom-end{
        border-top: $radii-8 solid $color-brand-white-base !important;
        bottom: -$space-16;
        right: calc(20% - var(--space-8));
    }
}

.tippy-arrow-dark {
    border: $radii-8 solid transparent;
    height: 0;
    position: absolute;
    width: 0;
    
    &-left{
        align-self: center;
        border-right: $radii-8 solid $color-brand-secondary-dark !important;
        left: -$space-16
    }

    &-left-start{
        align-self: flex-start;
        border-right: $radii-8 solid $color-brand-secondary-dark !important;
        left: -$space-16
    }

    &-left-end{
        align-self: flex-end;
        border-right: $radii-8 solid $color-brand-secondary-dark !important;
        left: -$space-16
    }
    
    &-right{
        align-self: center;
        border-left: $radii-8 solid $color-brand-secondary-dark !important;
        right: -$space-16
    }

    &-right-start{
        align-self: flex-start;
        border-left: $radii-8 solid $color-brand-secondary-dark !important;
        right: -$space-16
    }

    &-right-end{
        align-self: flex-end;
        border-left: $radii-8 solid $color-brand-secondary-dark !important;
        right: -$space-16
    }

    &-top{
        border-bottom: $radii-8 solid $color-brand-secondary-dark !important;
        left: calc(50% - var(--space-8));
        top: -$space-16;
        
    }

    &-top-start{
        border-bottom: $radii-8 solid $color-brand-secondary-dark !important;
        left: calc(20% - var(--space-8));
        top: -$space-16;
    }

    &-top-end{
        border-bottom: $radii-8 solid $color-brand-secondary-dark !important;
        right: calc(20% - var(--space-8));
        top: -$space-16;
    }
    
    &-bottom{
        border-top: $radii-8 solid $color-brand-secondary-dark !important;
        bottom: -$space-16;
        left: calc(50% - var(--space-8));
    }

    &-bottom-start{
        border-top: $radii-8 solid $color-brand-secondary-dark !important;
        bottom: -$space-16;
        left: calc(20% - var(--space-8));
    }

    &-bottom-end{
        border-top: $radii-8 solid $color-brand-secondary-dark !important;
        bottom: -$space-16;
        right: calc(20% - var(--space-8));
    }
}

.closeContainer{
    border-radius: $radii-50;
    cursor: pointer;
    height: $space-32;
    width: $space-32;
    &:hover{
        background: $color-brand-secondary-lighter;
        svg {
            path {
                stroke: $color-base-gray-20;
            }
        }
    }
}
.tippy-content {
    overflow-wrap: break-word;
}