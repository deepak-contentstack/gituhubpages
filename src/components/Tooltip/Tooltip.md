## **Tooltip**

A Tooltip Component can be used to indicate the user what is the functionality of certain element.

### **Tooltip Variant use**

- **dark**: Tooltip with dark variant can be used to show additional information for certain element.

  - props:
    - variant: dark

- **light**: Tooltip that has more details and components as a content (eg: Tooltip Card, Publish status.)

  - props:
    - variant: light

#### **Props**:

- **content**: String or React Component

  - required

- **position**: String

  - required
  - values : right | left | top | bottom

- **variantType**: String

  -optional (defaults to dark)
  -values : dark | light

- **showArrow**: Boolean

  - optional

- **trigger**: String

  - optional
  - default value : 'mouseenter focus'
  - other values: 'click', 'focusin', 'mouseenter click'

* **onHidden**: function
  Invoked once after tooltip hides.
  - optional


* **visible**: Boolean
  - Control the tooltips visibility
  - optional


* **showClose**: Boolean
  - Show a close icon at the top right corner of the tooltip
  - optional

* **onClose**: Functiom
  - Callback function to be executed after close icon is clicked
  - optional


