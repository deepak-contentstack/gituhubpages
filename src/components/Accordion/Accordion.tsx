import React, { useState, useMemo } from 'react';
import cn from 'classnames';

import withDeprecatedProp from '../../utils/hooks/depricatedPropsHoc';

// import Icon from '../Icon2/Icon';
//import Line from '../Line/Line';

import './Accordion.css';


export type accordionProps = {
  accordionLock?: boolean
  accordionData?: string
  addLeftComponent?: React.ReactNode
  addRightComponent?: React.ReactNode
  children?: React.ReactNode
  className?: string
  noChevron?: boolean
  preRender?: boolean
  renderExpanded?: boolean
  setTitleOnHide?: any | React.ReactNode
  title: string | React.ReactNode
  accordionDataCount?: number,
  hasBackgroundColor?: boolean
  /**
   * Actions List of {component: ReactNode, onClick: () => void}
   */
  actions?: AccordionActionsList,
  isAccordionOpen?: boolean | undefined
  onTitleClick?: () => void,
  errorMessage?: string
  dashedLineVisibility?: 'hover' | 'always',
  accordionTitlePropagation?: boolean
  testId?: string
};

type accordionTitleProps = {
  accordionLock?: boolean
  addLeftComponent?: React.ReactNode
  addRightComponent?: React.ReactNode
  accordionDataCount: number
  noChevron?: boolean
  setTitleOnHide: any | React.ReactNode | undefined
  title: string | React.ReactNode
  toggle: Function
  toggleStat?: boolean
  actions?: AccordionActionsList
  onClick?: () => void,
  error?: string
  hasBackgroundColor?: boolean
  dashedLineVisibility?: 'hover' | 'always',
  accordionTitlePropagation?: boolean
};

type accordionDataProps = {
  accordionData: string | React.ReactNode
};
type AccordionAction = {
  component: React.ReactNode
  onClick?: () => void,
  actionClassName?: string,
}

export type AccordionActionsList = AccordionAction[]

const AccordionTitle = (props: accordionTitleProps) => {
  // const noDefaultIcon = props.noDefaultIcon !== undefined ? props.noDefaultIcon : false;

  const getAccordionTitle = () => {
    if (!props.toggleStat) {
      return props.setTitleOnHide ? props.setTitleOnHide : props.title
    }
    else return props.title
  }

  const dashedLineClassname = cn('Accordion__heading__line', {
    'visible': props.dashedLineVisibility === 'always',
  });

  const { accordionTitlePropagation } = props || { accordionTitlePropagation: true }

  return useMemo(() => {

    const handleToggle = () => {
      if (!props.accordionLock) {
        if (props.onClick) props.onClick();
        props.toggle();
      }
    }

    return (
      <>
        <div className={`Accordion__heading ${props.accordionLock ? 'Accordion__heading-disabled' : ''}`} onClick={handleToggle}>
          <div className="flex">
            {props.addLeftComponent ? props.addLeftComponent : null}
            <div className={`Accordion__heading__title ${props.addLeftComponent ? 'ml-8' : ''} ${typeof props.title == 'string' ? 'Accordion__heading__title__ellipsis' : ''}`} onClick={(e) => { if (!accordionTitlePropagation) { e.stopPropagation() } }}>
              {getAccordionTitle()}
            </div>
          </div>
          {props.toggleStat === false && props.accordionDataCount ? <div className="Accordion__heading__field-count">{props.accordionDataCount}</div> : ""}
          {props.addRightComponent ? (
            props.addRightComponent
          ) : (
            <>
              {!props.noChevron &&
                <div className={props.accordionLock ? 'Accordion__heading__toggle-disabled' : 'Accordion__heading__toggle'}>
                </div>
              }
              <div className="flex-v-center flex-grow-1">
                {props.error && (
                  <div className="Accordion__heading__errorMsg">
                    {props.error}
                  </div>
                )}
                <span className={dashedLineClassname}>
                  {/* <Line type='dashed' /> */}
                </span>
              </div>
              {props.actions && <AccordionActions actions={props.actions} />}
            </>
          )}
        </div>
      </>
    );
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [
    props.toggleStat,
    props.accordionDataCount,
    props.title,
    props.error,
    props.addLeftComponent,
    props.addRightComponent,
  ])
};

export const AccordionData = (props: accordionDataProps) => {
  return <div className="Accordion__data">{props.accordionData}</div>
};

const AccordionActions = ({ actions }: {
  actions: AccordionAction[]
}) => {
  return (
    <div className="Accordion__actions">
      {actions?.map((action: AccordionAction, index) => {
        const actionClassname = cn("actions_list_item", action.actionClassName)
        return (
          <div
            className={actionClassname}
            onClick={(event) => { event.stopPropagation(); action.onClick(); }} key={index}
          >
            {action.component}
          </div>
        );
      })}
    </div>
  )
}

const Accordion = (props: accordionProps) => {

  const [toggleShow, setToggleShow] = useState(
    props.renderExpanded
  );

  const {
    title,
    accordionData,
    children,
    noChevron,
    accordionDataCount,
    hasBackgroundColor,
    dashedLineVisibility,
    accordionTitlePropagation,
    testId
  }: accordionProps = props;

  const [preRender, setPreRender] = useState(
    props.preRender
  );

  const accordionDataValue = accordionData !== undefined ? accordionData : children;

  const accordionDataCountValue = accordionDataCount;

  const toggleView = () => {
    setToggleShow(!toggleShow);
    if (preRender === false) {
      setPreRender(true);
    };
  };

  const classNames = cn('Accordion', props.className, {
    'Accordion--backgroundColor': hasBackgroundColor && !props.accordionLock,
  });

  let toggleState = toggleShow

  if (props.isAccordionOpen !== undefined) {
    toggleState = props.isAccordionOpen
  }

  return (
    <div className={classNames} data-test-id={testId}>
      <AccordionTitle
        title={title}
        setTitleOnHide={props.setTitleOnHide}
        toggleStat={toggleState}
        toggle={toggleView}
        hasBackgroundColor={hasBackgroundColor}
        noChevron={noChevron}
        addLeftComponent={props.addLeftComponent}
        addRightComponent={props.addRightComponent}
        accordionDataCount={accordionDataCountValue}
        accordionLock={props.accordionLock}
        actions={props.actions}
        onClick={props.onTitleClick}
        error={props.errorMessage}
        dashedLineVisibility={dashedLineVisibility}
        accordionTitlePropagation={accordionTitlePropagation}
      />
      {preRender === false ? (
        (toggleState === true) ? (
          <div className='Accordion__open'>
            <AccordionData accordionData={accordionDataValue} />
          </div>
        ) : null
      ) : (
        <div className={`${(toggleState) ? 'Accordion__open' : 'Accordion__collapsed'}`}>
          <AccordionData accordionData={accordionDataValue} />
        </div>
      )}
    </div>

  )
};

Accordion.defaultProps = {
  renderExpanded: false,
  preRender: false,
  noChevron: false,
  accordionLock: false,
  hasBackgroundColor: false,
  dashedLineVisibility: 'hover',
  accordionTitlePropagation: true,
  testId: 'cs-accordion'
} as Partial<accordionProps>;

export default withDeprecatedProp(
  Accordion,
  {
    'accordiondata': 'accordionData',
    'accordiondataCount': 'accordionDataCount',
    'accordionDataLock': 'accordionLock',
    'accordionState': 'isAccordionOpen',
    'backGroundColor': 'hasBackgroundColor',
    'errorMsg': 'errorMessage',
    'onClick': 'onTitleClick',
    'noDefaultIcon': 'noChevron',
    actions: [
      {
        className: 'actionClassName'
      }
    ]
  })
