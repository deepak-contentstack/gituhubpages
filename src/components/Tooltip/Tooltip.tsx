import './Tooltip.css'

import React, { createRef } from 'react'

import cn from 'classnames'

import Tippy, { TippyProps } from '@tippyjs/react'

// import Icon from '../Icon2'

enum ArrowDirection {
  "left-start" = "right-start",
  "left-end" = "right-end",
  "left" = "right",

  "right-start" = "left-start",
  "right-end" = "left-end",
  "right" = "left",

  "top-start" = "bottom-start",
  "top-end" = "bottom-end",
  "top" = "bottom",

  "bottom-start" = "top-start",
  "bottom-end" = "top-end",
  "bottom" = "top",
};

enum PaddingClass {
  "left-start" = "pl-10",
  "left-end" = "pl-10",
  "left" = "pl-10",

  "right-start" = "pr-10",
  "right-end" = "pr-10",
  "right" = "pr-10",

  "top-start" = "pt-10",
  "top-end" = "pt-10",
  "top" = "pt-10",

  "bottom-start" = "pb-10",
  "bottom-end" = "pb-10",
  "bottom" = "pb-10",
};
interface ITooltip extends TippyProps {
  content: string | React.ReactNode | { text: string; icon: any }
  position: string
  showArrow?: boolean
  onCreate?: (data: any) => void
  trigger?: string
  onHidden?: () => any
  disabled?: boolean
  variantType?: "dark" | "light" | "menuIcon"
  visible?: boolean,
  showClose?: boolean,
  onClose?: () => void,
  type?: "primary" | "secondary" | "dynamic",
  maxWidth?: any,
  style?: React.CSSProperties,
  [propsKey: string]: any
  testId?: string
  customTooltipCss?: string,
  wrapperElementType?: "div" | "span",
}

const Tooltip: React.FunctionComponent<ITooltip> = (props: any) => {
  const node = createRef<HTMLInputElement>()
  const { content, position, children, isMenuIcon, visible, showClose, testId, style, customTooltipCss, ...rest } = props

  const showArrow = props.showArrow !== undefined ? props.showArrow : false
  const variant = props.variantType || "dark"
  const tooltipType = props.type !== undefined ? props.type : "primary"
  const disabled = props.disabled ? props.disabled : false
  const interactive = typeof content === "string" ? false : true;

  let className =
    variant === "dark"
      ? "tippy-box-dark"
      : tooltipType === "primary"
        ? "tippy-box-light tippy-box-light--primary"
        : `tippy-box-light tippy-box-light--${tooltipType}`
  className = variant === "menuIcon" ? "tippy-box-light tippy-box-light-menu" : className
  className = customTooltipCss ? customTooltipCss : className
  className = cn(className, props.className);
  const wrapperClassNames = cn("tippy-wrapper", { [`${PaddingClass[position]}`]: showArrow });

  const getContent = () => {
    if (variant === "menuIcon") {
      return getKeyIcon(content.text, content.icon);
    }

    if (showArrow) {
      return (
        <div className="flex">
          {content}
          <div className={`tippy-arrow-${variant} tippy-arrow-${variant}-${ArrowDirection[position]}`} />
        </div >
      );
    }

    return content;
  }

  const wrapWithClose = (content) => {
    return (
      <>
        <div className="flex-v-center">
          <div>{content}</div>
          <div className="closeContainer flex-v-center flex-h-center" onClick={props.onClose ? () => props.onClose() : null}>
            {/* <Icon icon='Cancel' /> */}
          </div>
        </div>

      </>
    );
  };

  const WrapperTag = getWrapperType(props.wrapperElementType || "div");
  return (
    <Tippy
      content={showClose ? wrapWithClose(getContent()) : getContent()}
      className={className}
      placement={position}
      arrow={showArrow}
      interactive={interactive}
      disabled={disabled}
      visible={visible}
      {...rest}
    >
      <WrapperTag ref={node} className={wrapperClassNames} data-test-id={testId} style={style}>
        {children}
      </WrapperTag>
    </Tippy>
  )
}

const getKeyIcon = (text: string, icon: any) => {
  return (
    <div className="flex-v-center">
      <div>{text}</div>
      <div className="KeyShortcut">{icon}</div>
    </div>
  )
}
const getWrapperType = (type: string) => {
  if (type === "span") {
    return "span";
  }
  return "div";
}

Tooltip.defaultProps = {
  testId: "cs-tooltip",
  wrapperElementType: "div",
  trigger: "mouseenter focusin"
}

export default Tooltip
