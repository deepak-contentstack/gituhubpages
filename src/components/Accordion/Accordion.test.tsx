import React from 'react'
import renderer from 'react-test-renderer';
import { mount, shallow } from 'enzyme';

import Accordion, { AccordionData } from './Accordion'

// import { Icon } from '../Icon2/Icon'
import Tooltip from '../Tooltip/Tooltip'

describe('Accordion Component', () => {
  describe('Default Component', () => {
    it('renders the div component', () => {
      const output = renderer
        .create(<Accordion title="Default View" accordionData="This the the default accordion" />)
        .toJSON()
      expect(output).toMatchSnapshot()
    });

    it('renders the data of the Accordion on clicking on the title', () => {
      const wrapper = mount(<Accordion title="Default View" accordionData="This the the default accordion" />);
      expect(wrapper.find('.Accordion__data').length).toBe(0);
      wrapper.find('.Accordion__heading').simulate('click');
      expect(wrapper.find('.Accordion__data').length).toBe(1);
    });

    it('It should render the title as a component', () => {
      const DummyTitle = () => {
        return <span className="DummyTitle">Render Title As Component</span>
      }
      const wrapper = mount(<Accordion title={<DummyTitle />} accordionData="This the the default accordion" />);
      expect(wrapper.find('.DummyTitle').length).toBe(1);
    });

    it('Render the accordion as children', () => {
      const wrapper = mount(<Accordion title="Default View" >Render Accordion as Children </Accordion>);
      expect(wrapper.find('.Accordion__data').length).toBe(0);
      wrapper.find('.Accordion__heading').simulate('click');
      expect(wrapper.find('.Accordion__data').length).toBe(1);
      expect(wrapper.find('.Accordion__data').html()).toContain('Render Accordion as Children');
    });

    it('Render the accordion as open', () => {
      const wrapper = mount(<Accordion title="Default View" isAccordionOpen={true} accordionData="This the the default accordion" />);
      expect(wrapper.find('.Accordion__data').length).toBe(1);
    });

    it('Render the accordion as open an disabled', () => {
      const wrapper = mount(<Accordion title="Default View" accordionLock={true} renderExpanded={true} accordionData="This the the default accordion" />);
      expect(wrapper.find('.Accordion__data').length).toBe(1);
      wrapper.find('.Accordion__heading').simulate('click');
      expect(wrapper.find('.Accordion__data').length).toBe(1);
    });

    it('When the accordion is closed and locked, it should not allow the user to see the data', () => {
      const wrapper = mount(<Accordion title="Default View" accordionLock={true} accordionData="This the the default accordion" />);
      expect(wrapper.find('.Accordion__data').length).toBe(0);
      wrapper.find('.Accordion__heading').simulate('click');
      expect(wrapper.find('.Accordion__data').length).toBe(0);
    });

    it('It should preRender the data of the accordion', () => {
      const wrapper = mount(<Accordion title="Default View" preRender={true} accordionLock={true} accordionData="This the the default accordion" />);
      expect(wrapper.find('.Accordion__data').length).toBe(1);
    });

    it('It should render the right component', () => {
      const DummyComponent = () => {
        return (
          <span
            className="DummyComponent"
            onClick={(event: any) => {
              event.stopPropagation()
              alert('Right Component Clicked')
            }}
          >
          </span>
        )
      }
      const wrapper = mount(<Accordion title="Default View" addRightComponent={<DummyComponent />} accordionData="This the the default accordion" />);
      expect(wrapper.find('.DummyComponent').length).toBe(1);
    });

    it('It should render the left component', () => {
      const DummyComponent = () => {
        return (
          <span
            className="DummyComponent"
            onClick={(event: any) => {
              event.stopPropagation()
              alert('Right Component Clicked')
            }}
          >
          </span>
        )
      }
      const wrapper = mount(<Accordion title="Default View" addLeftComponent={<DummyComponent />} accordionData="This the the default accordion" />);
      expect(wrapper.find('.DummyComponent').length).toBe(1);
    });

    it('It should render the error text for the component', () => {
      const wrapper = mount(<Accordion title="Default View" errorMessage="This is the error message" accordionData="This the the default accordion" />);
      expect(wrapper.find('.Accordion__heading__errorMsg').html()).toContain('This is the error message');
    });

    it('It should render the data count of the accordion when the accordion is closed', () => {
      const wrapper = mount(<Accordion title="Default View" accordionDataCount={2} accordionData="This the the default accordion" />);
      expect(wrapper.find('.Accordion__heading__field-count').html()).toContain('2');
      wrapper.find('.Accordion__heading').simulate('click');
      expect(wrapper.find('.Accordion__heading__field-count').debug()).toContain('');
    });

    it('It should render custom actions', () => {
      const actions: any = [
        {
          component: (
            <Tooltip content="Delete" position="top" showArrow={false}></Tooltip>
          ),
          onClick: () => {
            alert('Delete triggered')
          },
          actionClassName: "ActionListItem--warning",
        },
        {
          component: (
            <Tooltip content="Edit" position="top" showArrow={false}></Tooltip>
          ),
          onClick: () => {
            alert('Edit triggered')
          },
          actionClassName: "ActionListItem--warning",
        },
      ]
      const wrapper = mount(<Accordion title="Default View" actions={actions} accordionData="This the the default accordion" />);
      expect(wrapper.find('.Accordion__actions').length).toBe(1);
      expect(wrapper.find('.ActionListItem--warning').length).toBe(2);
    });

    it('It should render custom actions', () => {

      const mockFunction = jest.fn();

      const actions: any = [
        {
          component: (
            <Tooltip content="Delete" position="top" showArrow={false}></Tooltip>
          ),
          onClick: mockFunction,
          actionClassName: "ActionListItem--warning",
        },
        {
          component: (
            <Tooltip content="Edit" position="top" showArrow={false}></Tooltip>
          ),
          onClick: () => {
            alert('Edit triggered')
          }
        },
      ]
      const wrapper = mount(<Accordion title="Default View" actions={actions} accordionData="This the the default accordion" />);
      wrapper.find('.ActionListItem--warning').simulate('click');
      expect(mockFunction).toBeCalledTimes(1);
    });

    it('It should set the title on hide', () => {
      const wrapper = mount(<Accordion title="Default View" setTitleOnHide="On Hide Title" accordionData="This the the default accordion" />);
      expect(wrapper.find('.Accordion__heading__title').html()).toContain('On Hide Title');
      wrapper.find('.Accordion__heading').simulate('click');
      expect(wrapper.find('.Accordion__heading__title').html()).toContain('Default View');
      wrapper.find('.Accordion__heading').simulate('click');
      expect(wrapper.find('.Accordion__heading__title').html()).toContain('On Hide Title');
    });

    it('It should not propagate the click on the title', () => {
      const wrapper = mount(<Accordion title="Default View" accordionTitlePropagation={false} setTitleOnHide="On Hide Title" accordionDataCount={2} accordionData="This the the default accordion" />);
      expect(wrapper.find('.Accordion__heading__title').html()).toContain('On Hide Title');
      wrapper.find('.Accordion__heading__title').simulate('click');
      expect(wrapper.find('.Accordion__heading__title').html()).toContain('On Hide Title');
    });

    it('It should trigger the function on clicking the title', () => {
      const MockFunction = jest.fn();
      const wrapper = mount(<Accordion title="Default View" onTitleClick={MockFunction} accordionData="This the the default accordion" />);
      wrapper.find('.Accordion__heading').simulate('click');
      expect(MockFunction).toBeCalledTimes(1);
    });

    it('Should not render the chevron when noChevron is true', () => {
      const wrapper = mount(<Accordion title="Default View" noChevron={true} accordionData="This the the default accordion" />);
      expect(wrapper.find('.Accordion__heading__toggle').length).toBe(0);
    });
  })
})
