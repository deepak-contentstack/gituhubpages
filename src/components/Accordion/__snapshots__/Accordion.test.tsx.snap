// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Component Default Component renders the div component 1`] = `
<div
  className="Accordion"
  data-test-id="cs-accordion"
>
  <div
    className="Accordion__heading "
    onClick={[Function]}
  >
    <div
      className="flex"
    >
      <div
        className="Accordion__heading__title  Accordion__heading__title__ellipsis"
        onClick={[Function]}
      >
        Default View
      </div>
    </div>
    
    <div
      className="Accordion__heading__toggle"
    />
    <div
      className="flex-v-center flex-grow-1"
    >
      <span
        className="Accordion__heading__line"
      />
    </div>
  </div>
</div>
`;
